AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine+DeleteTransaction.swift

```diff
@@ -32,10 +32,10 @@ extension StorageEngine {
 
     func queryAndDeleteTransaction<M: Model>(_ modelType: M.Type,
```

Comment:
where is this function used?

Reply:
this is used by the StorageEngine's delete API, so whenever a model is deleted, we perform a cascade delete by finding the associated children objects as well, deleting all models locally in a transaction, and then syncing them to the cloud